Added placeholder "Add key" for a new tag key

bhavyaKhatri2703 opened the pull request:
This pr adds a placehoder text to the empty key field which is for adding a new tag
<img width="386" height="261" alt="image" src="https://github.com/user-attachments/assets/48b00998-f91f-4cd3-af32-27afb7fd5c58" />
Fix #11211
Pls let me know if need to make some other changes.

bhavyaKhatri2703 committed:
Added placeholder for new tag key

tyrasd committed:
make "add new tag" text translatable

tyrasd committed:
fix rounded corner of tag list

tyrasd committed:
disable info-(i) button for "add new tag" row

tyrasd committed:
fix indentation, drop not really necessary inline comments

tyrasd committed:
add to changelog

tyrasd reviewed (approved):
Thanks for the PR. 😊  The only thing that was missing was to make the text translatable (see [documentation](https://github.com/openstreetmap/iD/wiki/How-to-Use-iD%27s-Localizer-Module)). I went ahead and added that in f0e9e9624. I also added some additional commits to this branch that address some additional details raised in https://github.com/openstreetmap/iD/issues/11211#issuecomment-3072880828.

tyrasd committed:
Merge branch 'develop' into tag-key-placeholder

bhavyaKhatri2703 commented:
> Thanks for the PR. 😊 The only thing that was missing was to make the text translatable (see [documentation](https://github.com/openstreetmap/iD/wiki/How-to-Use-iD%27s-Localizer-Module)). I went ahead and added that in [f0e9e96](https://github.com/openstreetmap/iD/commit/f0e9e962401e224535a8bb93a6d8e19158099b1f). I also added some additional commits to this branch that address some additional details raised in [#11211 (comment)](https://github.com/openstreetmap/iD/issues/11211#issuecomment-3072880828).

 thank you!
